de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/util/CollaborationUtils.java

```diff
@@ -303,70 +290,75 @@ private static String getSessionDescription(ISarosSession sarosSession) {
+      resources.removeAll(ResourceAdapterFactory.convertBack(session.getSharedResources()));
     }
 
-    final int resourcesSize = selectedResources.size();
+    final int resourcesSize = resources.size();
 
-    IResource[] preSortedResources = new IResource[resourcesSize];
+    final IResource[] preSortedResources = new IResource[resourcesSize];
 
     int frontIdx = 0;
     int backIdx = resourcesSize - 1;
 
     // move projects to the front so the algorithm is working as expected
-    for (IResource resource : selectedResources) {
+    for (final IResource resource : resources) {
       if (resource.getType() == IResource.PROJECT) preSortedResources[frontIdx++] = resource;
       else preSortedResources[backIdx--] = resource;
     }
 
-    Map<IProject, Set<IResource>> projectsResources = new HashMap<IProject, Set<IResource>>();
+    final Map<IProject, Set<IResource>> projectsToResourcesMapping =
+        new HashMap<IProject, Set<IResource>>();
 
-    for (IResource resource : preSortedResources) {
+    for (final IResource resource : preSortedResources) {
+
+      // init full shared projects
       if (resource.getType() == IResource.PROJECT) {
-        projectsResources.put((IProject) resource, null);
+        projectsToResourcesMapping.put(resource.getAdapter(IProject.class), null);
         continue;
       }
 
-      IProject project = resource.getProject();
+      final IProject project = resource.getProject();
 
       if (project == null) continue;
 
-      if (!projectsResources.containsKey(project))
-        projectsResources.put(project, new HashSet<IResource>());
+      // init partial shared projects
+      if (!projectsToResourcesMapping.containsKey(project))
+        projectsToResourcesMapping.put(project, new HashSet<IResource>());
 
-      Set<IResource> resources = projectsResources.get(project);
+      final Set<IResource> projectResources = projectsToResourcesMapping.get(project);
 
-      // if the resource set is null, it is a full shared project
-      if (resources != null) resources.add(resource);
+      // if the project resource set is null, it is a full shared project
+      if (projectResources != null) projectResources.add(resource);
     }
 
-    List<IResource> additionalFilesForPartialSharing = new ArrayList<IResource>();
+    for (Entry<IProject, Set<IResource>> entry : projectsToResourcesMapping.entrySet()) {
 
-    for (Entry<IProject, Set<IResource>> entry : projectsResources.entrySet()) {
+      final IProject project = entry.getKey();
+      final Set<IResource> projectResources = entry.getValue();
 
-      IProject project = entry.getKey();
-      Set<IResource> resources = entry.getValue();
+      if (projectResources == /* full shared */ null) continue;
 
-      if (resources == /* full shared */ null) continue;
+      // do not add the files again in case this is an already partial shared project
+      if (session != null && session.isShared(ResourceAdapterFactory.create(project))) continue;
```

Comment:
@stefaus Was you aware of this logic when you added your InstantSessionStart feature ? Mainly line 365.

Reply:
yes, I just prio them if they are included in the negotiation file list (so same file list as in Archive transmission)